ci-operator/step-registry/ipi/conf/vsphere/check/vcm/ipi-conf-vsphere-check-vcm-commands.sh

```diff
@@ -0,0 +1,442 @@
+}
+
+if [[ ${JOB_NAME_SAFE} =~ "-upi" ]]; then
+   IPI=0
+   log "determined this is a UPI job"
+else
+   IPI=1
+   log "determined this is an IPI job"
+fi
+
+if [[ -n "${VSPHERE_EXTRA_LEASED_RESOURCE:-}" ]]; then
+  log "creating extra lease resources"
+
+  i=1  
+  for extra_leased_resource in ${VSPHERE_EXTRA_LEASED_RESOURCE}; do
+    log "creating extra leased resource ${extra_leased_resource}"
+    echo "apiVersion: vspherecapacitymanager.splat.io/v1
+kind: Lease
+metadata:
+  generateName: "${LEASED_RESOURCE}-"
+  namespace: "vsphere-infra-helpers"
+  annotations: {}
+  labels:
+    boskos-lease-id: "${LEASED_RESOURCE}"
+    job-name: "${JOB_NAME_SAFE}"
+    VSPHERE_EXTRA_LEASED_RESOURCE: \"${i}\"
+spec:
+  vcpus: 0
+  memory: 0
+  networks: 1" | oc create --kubeconfig ${SA_KUBECONFIG} -f -
+
+  i=$((i + 1))
+  done
+fi
+
+
+if [[ -n "${VSPHERE_BASTION_LEASED_RESOURCE:-}" ]]; then  
+  log "creating bastion lease resource ${VSPHERE_BASTION_LEASED_RESOURCE}"
+  echo "apiVersion: vspherecapacitymanager.splat.io/v1
+kind: Lease
+metadata:
+  generateName: "${LEASED_RESOURCE}-"
+  namespace: "vsphere-infra-helpers"
+  annotations: {}
+  labels:
+    boskos-lease-id: "${LEASED_RESOURCE}"
+    job-name: "${JOB_NAME_SAFE}"
+    VSPHERE_BASTION_LEASED_RESOURCE: \"${VSPHERE_BASTION_LEASED_RESOURCE}\"
+spec:
+  vcpus: 0
+  memory: 0
+  requiresPool: \"${VSPHERE_BASTION_LEASED_RESOURCE}\"
+  networks: 1" | oc create --kubeconfig ${SA_KUBECONFIG} -f -
+fi
+
+
+POOLS=${POOLS:-}
+declare -a pools=($POOLS)
+
+SA_KUBECONFIG=${SA_KUBECONFIG:-/var/run/vault/vsphere-ibmcloud-ci/vsphere-capacity-manager-kubeconfig}
+OPENSHIFT_REQUIRED_CORES=${OPENSHIFT_REQUIRED_CORES:-24}
+OPENSHIFT_REQUIRED_MEMORY=${OPENSHIFT_REQUIRED_MEMORY:-96}
+
+if [[ ${#pools[@]} -eq 0 ]]; then
+  pools[0]="unspecified"
+else
+  # if we have multiple pools, attempt to spread the load evenly between the pools
+  OPENSHIFT_REQUIRED_CORES=$((OPENSHIFT_REQUIRED_CORES / ${#pools[@]}))
+  OPENSHIFT_REQUIRED_MEMORY=$((OPENSHIFT_REQUIRED_MEMORY / ${#pools[@]}))
+fi
+
+# create a lease for each pool
+for POOL in ${pools[@]}; do
+  log "creating lease for pool ${POOL}"
+  requiredPool=""
+  if [ $POOL != "unspecified" ]; then 
+    requiredPool="required-pool: $POOL"
+    log "setting required pool ${requiredPool}"
+  fi
+  echo "apiVersion: vspherecapacitymanager.splat.io/v1
+kind: Lease
+metadata:
+  generateName: "${LEASED_RESOURCE}-"
+  namespace: "vsphere-infra-helpers"
+  annotations: {}
+  labels:
+    boskos-lease-id: "${LEASED_RESOURCE}"
+    job-name: "${JOB_NAME_SAFE}"
+spec:
+  vcpus: ${OPENSHIFT_REQUIRED_CORES}
+  memory: ${OPENSHIFT_REQUIRED_MEMORY}
+  ${requiredPool}
+  networks: 1" | oc create --kubeconfig ${SA_KUBECONFIG} -f -
+done
+
+log "waiting for lease to be fulfilled..."
+oc wait leases.vspherecapacitymanager.splat.io --kubeconfig ${SA_KUBECONFIG} --timeout=30m --for=jsonpath='{.status.phase}'=Fulfilled -n vsphere-infra-helpers -l boskos-lease-id="${LEASED_RESOURCE}"
```

Comment:
default is 30 seconds, what happens if there are no resources available in a pool?

Reply:
nm lol